src/library/dao/trackdao.cpp

```diff
@@ -2064,11 +2092,11 @@ TrackPointer TrackDAO::getOrAddTrack(const QString& trackLocation,
                                      bool processCoverArt,
                                      bool* pAlreadyInLibrary) {
     TrackId trackId(getTrackId(trackLocation));
-    bool track_already_in_library = trackId.isValid();
+    bool trackAlreadyInLibrary = trackId.isValid();
 
     // Add Track to library -- unremove if it was previously removed.
     if (!trackId.isValid()) {
-        trackId = addTrack(trackLocation, true);
+        trackId = addSingleTrack(trackLocation, true);
```

Comment:
Here is room for optimization. We know that the track is new and we already head a TrackInfo Object inside of addSingleTrack. Can't we just add it to the cache?